pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/CastTransformFunction.java

```diff
@@ -191,7 +194,19 @@ private long[] transformToTimestampValuesSV(ValueBlock valueBlock) {
       int length = valueBlock.getNumDocs();
       initLongValuesSV(length);
       String[] stringValues = _transformFunction.transformToStringValuesSV(valueBlock);
-      ArrayCopyUtils.copyToTimestamp(stringValues, _longValuesSV, length);
+      RoaringBitmap nullBitmap = _transformFunction.getNullBitmap(valueBlock);
+      if (nullBitmap != null) {
```

Comment:
Do we need to check if it is empty? (not sure if empty one is returned as `null`)

Reply:
Yeah, empty is returned as `null` - https://github.com/apache/pinot/blob/0ea5bceafaada90598af85d6d8b71041cffea6d6/pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/BaseTransformFunction.java#L791-L793

https://github.com/apache/pinot/blob/0ea5bceafaada90598af85d6d8b71041cffea6d6/pinot-core/src/main/java/org/apache/pinot/core/operator/docvalsets/ProjectionBlockValSet.java#L67-L79